Set/Set/SetGame.swift

```diff
@@ -7,37 +7,53 @@
 
 import Foundation
 
+enum CardState {
+    case chosen, matched, unmatched
+}
+
 final class SetGame {
 
     var canDrawCards: Bool {
-        self.shownCards.count < 24
+        (self.shownCards.compactMap { $0 }.count < 24 ||
+         self.matchedCardsIndices.count > 2) &&
+        self.deck.count > 2
     }
+
     private var deck = Array<Card>()
-    private var shownCards: Array<Card>
+    private var shownCards: Array<Card?>
     private var chosenCardsIndices: Array<Int>
     private var matchedCardsIndices: Array<Int>
+    private var unmatchedCardsIndices: Array<Int>
+    private(set) var score: Int
 
     init() {
-        self.shownCards = []
+        self.shownCards = .init(repeating: nil, count: 24)
         self.chosenCardsIndices = []
         self.matchedCardsIndices = []
+        self.unmatchedCardsIndices = []
+        self.score = 0
         generateDeck()
     }
 
-    func drawCards(_ amount: Int) {
-        for _ in 0..<amount {
-            drawOneCard()
+    func dealCards(_ amount: Int) {
+        if matchedCardsIndices.count > 2 {
```

Comment:
can we do the check against `3`? Doesn't makes anyhow better check for `< 2` unless I'm missing some corner case

Reply:
that's just for your "can we make this for any amount of cards"